File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const checked = new Map();
        for(let i = 0; i < nums.length; i++) {
            let diff = target - nums[i];
            
            if(checked.has(diff)){
                return [checked.get(diff), i];
            } else {
                checked.set(nums[i] ,i);
            }
        }
        return [];
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        l = 0
        r = len(nums) - 1

        while l < r:
            twoSum = nums[l] + nums[r]
            if twoSum < target:
                l += 1
            elif twoSum > target:
                r -= 1
            else:
                return [l+1,r+1]